### PR TITLE
ISPN-10261 Remove Total Order Transaction support

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.redhat.com/browse/ISPN-10261

Pending:
- [x] Documentation
- [x] Remove DeliverOrder.TOTAL?
- [x] Check for optimizations (optional)

### CODE REVIEW

File: core/src/main/java/org/infinispan/configuration/cache/Configurations.java [hunk 1]
```diff
@@ -39,16 +39,6 @@ public static boolean isOnePhaseCommit(Configuration cfg) {
             cfg.transaction().lockingMode() == LockingMode.PESSIMISTIC;
    }
 
-   public static boolean isOnePhaseTotalOrderCommit(Configuration cfg) {
```
Comment (wburns) [inline core/src/main/java/org/infinispan/configuration/cache/Configurations.java]: Believe we have to deprecate this method first as it is in a public package. Should be able to just return false always I assume.

Comment (wburns) [reply core/src/main/java/org/infinispan/configuration/cache/Configurations.java]: Or maybe this method is obscure enough that we don't need to worry.

Comment (pruivo) [reply core/src/main/java/org/infinispan/configuration/cache/Configurations.java]: hmmm.. I assumed this class was internal. I don't see any use case for a user to use any of this methods :/ 
If you want, I can add it back.

Comment (wburns) [reply core/src/main/java/org/infinispan/configuration/cache/Configurations.java]: It is in a package that is public. We should probably just put it back in and always return false and deprecate for now.

Comment (pruivo) [reply core/src/main/java/org/infinispan/configuration/cache/Configurations.java]: done!

**Comment by wburns** [comment]
Integrated into master, thanks @pruivo !

**Pull request merged**